containers/podman-desktop
# feat: switch to using libpodapi for listImages

feat: switch to using libpodapi for listImages

### What does this PR do?

* Switches to now using podmanListImages for listing images, using the
  libpodapi, falling back to compat if not available
* Adds a configuration value that's hidden that can be easily changed
  if we encounter issues in the future with regards to isManifest.
* Hides any manifest images appearing in the UI, HOWEVER, you can still
  get the manifests with the podman desktop api

### Screenshot / video of UI



N/A, you should not notice any UI change, but instead if you call

window.listImages(), you should get the manifests in the list

### What issues does this PR fix or reference?



Closes https://github.com/containers/podman-desktop/issues/6676
Closes https://github.com/containers/podman-desktop/issues/6689
Closes https://github.com/containers/podman-desktop/issues/6622

### How to test this PR?




You can also test this via:

Building a manifest:

```
podman build --platform linux/arm64,linux/amd64 --manifest testmanifest2 .
```

And then:

1. Going to Images within PD and seeing that your manifest is NOT
   listed.
2. Adding console.log(window.listImages()) somewhere in your code you
   will be able to see that manifests are being returned now in the
   listImages() API call.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>
